shell/pages/c/_cluster/explorer/index.vue

```diff
@@ -387,7 +438,27 @@ export default {
         return !conditions ? acc : acc + 1;
       }, 0);
 
-      return count === 0;
+      if (!count) {
+        return SERVICE_STATUS.UNHEALTHY;
+      }
+
+      return SERVICE_STATUS.HEALTHY;
+    },
+
+    getAgentStatus(agent, disconnected = false) {
+      if (!agent) {
```

Comment:
The `disconnected === true` case is valid only for `cattle` agent, so it needs to be outside the `getAgentStatus`, otherwise I would bind a agent id decode (`cattle/fleet`) in the function itself .